galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager.ivt/src/main/java/dev/galasa/cicsts/manager/ivt/CICSTSManagerIVT.java

```diff
@@ -0,0 +1,199 @@
+	   } catch (CicstsManagerException e) {
+		   exceptionThrown = true;
+	   }
+	   assertThat(exceptionThrown).isFalse();
+	   assertThat(regionJob).isNotNull();	   
+   }
+   
+   /**
+    * Tests that the CICS Terminal in the CICS TS Manager retrieves the correct CICS Region
+    * @throws FieldNotFoundException 
+    * @throws NetworkException 
+    * @throws TerminalInterruptedException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test
+   public void testGetCicsRegion() throws TimeoutException, KeyboardLockedException, TerminalInterruptedException, NetworkException, FieldNotFoundException {
+	   logger.info("Testing that the CICS Terminal gets the correct CICS Region");
+	   String testCicsRegion = terminal.getCicsRegion().toString().replace("CICS Region[", "").replace("]", "");
+	   logger.info("CICS Region retrieved by the ICicsTerminal: " + testCicsRegion);
+	   terminal.type("CEMT INQUIRE").enter().waitForKeyboard();
+	   assertThat(terminal.retrieveScreen().contains(testCicsRegion)).isTrue();
+   }
+   
+   /**
+    * Tests that the CICS Terminal in the CICS TS Manager correctly connects to the CICS Region
+    * @throws CicstsManagerException
+    * @throws TerminalInterruptedException 
+    * @throws NetworkException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test
+   public void testConnectToCicsRegion() throws CicstsManagerException, TerminalInterruptedException, KeyboardLockedException, NetworkException, TimeoutException {
+	   logger.info("Testing that the CICS Terminal connects to the CICS Region");
+	   logger.info("First, need to disconnect from the CICS Region");
+	   terminal.disconnect();
+	   assertThat(terminal.isConnected()).isFalse();
+	   terminal.connectToCicsRegion();
+	   assertThat(terminal.isConnected()).isTrue();
+	   assertThat(terminal.isClearScreen()).isTrue();
+   }
+   
+   /**
+    * Tests that the CICS Terminal in the CICS TS Manager resets and clears correctly
+    * @throws FieldNotFoundException 
+    * @throws NetworkException 
+    * @throws TerminalInterruptedException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test
+   public void testResetAndClear() throws TimeoutException, KeyboardLockedException, TerminalInterruptedException, NetworkException, FieldNotFoundException {
+	   logger.info("Testing that the CICS Terminal resets and clears screen correctly");
+	   boolean exceptionThrown = false;
+	   try {
+		   terminal.resetAndClear();
+	   } catch (CicstsManagerException e) {
+		   exceptionThrown = true;
+	   }
+	   assertThat(exceptionThrown).isFalse();
+	   assertThat(terminal.isClearScreen()).isTrue();
+   }
+   
+   /**
+    * Tests that the CICS Terminal in the CICS TS Manager sets uppercase translation correctly
+    * @throws CicstsManagerException 
+    */
+   @Test
+   public void testUppercaseTranslation() throws CicstsManagerException {
```

Comment:
We should validate that lower case characters are upper cased correctly.  Log onto the CICS region and use the transaction CEOT TRANID that allows for mixed case/ upper case translation and add content to this test to check that things are upper cases

Reply:
I have defined a lower case variable with uppercase translation on and checked that when retrieved it was upper cased. Similarly I have defined a lower case variable with uppercase translation off and checked that when retrieved it is unchanged.